### PR TITLE
java: Add option to include method modifiers

### PR DESCRIPTION
**Pull request by mpozniak95**
## Description
Added an option for java profiling: --java-include-method-modifiers, disabled by default.
When flag provided we use: jvmti->GetMethodModifiers in async-profiler to get method modifiers.

## Related Issue
https://github.com/Granulate/gprofiler/issues/570

## How Has This Been Tested?
Added test that tests this case

### CODE REVIEW

**Comment by mpozniak95** [comment]
Needs this: https://github.com/Granulate/async-profiler/pull/5 to be merged first, and then need to make a change in async_profiler_shared_build.sh

File: gprofiler/profilers/java.py
Comment (Jongy) [inline gprofiler/profilers/java.py:489]: Make it a oneliner?

**Review by Jongy — CHANGES_REQUESTED** [review]
One small comment here, fix the AP-side comments and then update the branch so the test matches what I requested

File: scripts/async_profiler_build_shared.sh
Comment (mpozniak95) [inline scripts/async_profiler_build_shared.sh:13]: I had to add `-std=c++11` because build was failing on centos glibc builder. It seems that by default centos gcc compile with c++98 standard. I am not sure if that change is ok?

File: tests/test_java.py
Comment (Jongy) [inline tests/test_java.py:1150]: Good negative test.

File: scripts/async_profiler_build_shared.sh
Comment (Jongy) [reply scripts/async_profiler_build_shared.sh:13]: Ideally we should not make wide changes in the build such as modifying default flags. I'm not sure if it has other side effects on AP's built code. Can you do it without? i.e if the problem is `const std::vector<std::pair<int, std::string>> access_flags` then use a different construct and not a vector of pairs?

Comment (Jongy) [reply scripts/async_profiler_build_shared.sh:13]: I prefer the changes we do on upstream to remain local, i.e when the includemm option is not given, the behavior should be exact. If we change the global build flags, it's maybe not exact anymore :/

File: scripts/async_profiler_build_shared.sh
Comment (Jongy) [inline scripts/async_profiler_build_shared.sh:11]: Can revert the changes here

Comment (Jongy) [reply scripts/async_profiler_build_shared.sh:11]: (I mean, reminder to do so after we merge the other)

**Comment by Jongy** [comment]
Can update the commit, and we're good :)

**Comment by Jongy** [comment]
Pushed tag `v2.9g6`

File: scripts/async_profiler_build_shared.sh
Comment (Jongy) [inline scripts/async_profiler_build_shared.sh:12]:
```suggestion
```

Comment (mpozniak95) [reply scripts/async_profiler_build_shared.sh]: Sorry for that :(

**Comment by Jongy** [comment]
Looks good, I will merge once tests pas..

**Review by Jongy — APPROVED** [review]

**Pull request merged**
